editor/src/main/com/mbrlabs/mundus/utils/Log.java

```diff
@@ -56,110 +62,223 @@ public void uncaughtException(Thread thread, Throwable throwable) {
+        if (LOG_LEVEL >= DEBUG) {
+            msg = completeMsg(msg, params);
+            print("[Debug][" + tag + "] " + msg);
+        }
     }
 
-    public static void info (String tag, String msg) {
-        if (LOG_LEVEL >= INFO) print("[Info][" + tag + "] " + msg);
+    public static void infoTag(String tag, String msg, Object... params) {
+        if (LOG_LEVEL >= INFO) {
+            msg = completeMsg(msg, params);
+            print("[Info][" + tag + "] " + msg);
+        }
     }
 
-    public static void warn (String tag, String msg) {
-        if (LOG_LEVEL >= WARN) print("[Warning][" + tag + "] " + msg);
+    public static void warnTag(String tag, String msg, Object... params) {
+        if (LOG_LEVEL >= WARN) {
+            msg = completeMsg(msg, params);
+            print("[Warning][" + tag + "] " + msg);
+        }
     }
 
-    public static void error (String tag, String msg) {
-        if (LOG_LEVEL >= ERROR) printErr("[Error][" + tag + "] " + msg);
+    public static void errorTag(String tag, String msg, Object... params) {
+        if (LOG_LEVEL >= ERROR) {
+            msg = completeMsg(msg, params);
+            printErr("[Error][" + tag + "] " + msg);
+        }
     }
 
-    public static void fatal (String tag, String msg) {
-        if (LOG_LEVEL >= FATAL) printErr("[Fatal][" + tag + "] " + msg);
+    public static void fatalTag(String tag, String msg, Object... params) {
+        if (LOG_LEVEL >= FATAL) {
+            msg = completeMsg(msg, params);
+            printErr("[Fatal][" + tag + "] " + msg);
+        }
     }
 
-    private static void print (String msg) {
+    private static void print(String msg) {
         msg = getTimestamp() + msg;
         logFileWriter.println(msg);
         logFileWriter.flush();
         System.out.println(msg);
     }
 
-    private static void printErr (String msg) {
+    private static void printErr(String msg) {
         msg = getTimestamp() + msg;
         System.err.println(msg);
     }
 
-    public static void exception (Throwable e) {
+    public static void exception(Throwable e) {
         String stack = ExceptionUtils.getStackTrace(e);
         fatal(stack);
     }
 
-    private static String getTimestamp () {
+    private static String getTimestamp() {
         return msgDateFormat.format(new Date());
     }
 
     private static class ErrorStreamInterceptor extends PrintStream {
-        public ErrorStreamInterceptor (OutputStream out) {
+
+        public ErrorStreamInterceptor(OutputStream out) {
             super(out, true);
         }
 
         @Override
-        public void print (String s) {
+        public void print(String s) {
             super.print(s);
-            if (logFileWriter != null) logFileWriter.println(s);
+            if (logFileWriter != null) {
+                logFileWriter.println(s);
+            }
+        }
+    }
+
+    private static String completeMsg(String msg, Object... params) {
```

Comment:
Here you can ditch the temp string. Also ""+p is not a good practice
msg = msg.replaceFirst("\\{\\}", String.valueOf(p));